chalice/app.py

```diff
@@ -15,31 +15,39 @@
 mtx_handler = LoomMatrixHandler()
 
 
+def api_endpoint_decorator(api_endpoint_func):
+    @wraps(api_endpoint_func)
+    def api_endpoint_wrapper(*args, **kwargs):
+        try:
+            return api_endpoint_func(*args, **kwargs)
+        except KeyError:
```

Comment:
What are you trying to catch here?

Reply:
If the `api_endpoint_func()` is `check_request_status()`, then an `KeyError` exception indicates that the request has not been made before.

Reply:
Ah gotcha, that makes sense. In the future we might want to make a special exception for that, but don't need to worry about that for now.